packages/core/src/sync-historical/index.test.ts

```diff
@@ -46,22 +47,28 @@ beforeEach(setupCleanup);
 
 const setupChildAddresses = (
   sources: Source[],
-): Map<Factory, Map<Address, number>> => {
+): Map<string, Map<Address, number>> => {
```

Comment:
Another idea to consider - you could export a type
```
export type FactoryId = string;
```
from the `factory.ts` file, and use it instead of `string` here to make it easier to understand at a glance. Lightweight version of [branded types](https://egghead.io/blog/using-branded-types-in-typescript). Not a strong opinion though.